src/analysis/inner_product_space/orthogonal.lean

```diff
@@ -200,3 +205,148 @@ lemma orthogonal_family_self :
 | ff ff := absurd rfl
 
 end submodule
+
+@[simp]
+lemma bilin_form_of_real_inner_orthogonal {E} [normed_add_comm_group E] [inner_product_space ℝ E]
+  (K : submodule ℝ E) :
+  bilin_form_of_real_inner.orthogonal K = Kᗮ := rfl
+
+/-!
+### Orthogonality of submodules
+
+In this section we define `submodule.is_ortho U V`, with notation `U ⟂ V`.
+
+The API roughly matches that of `disjoint`.
+-/
+namespace submodule
+
+/-- The proposition that two submodules are orthogonal. Has notation `U ⟂ V`. -/
+def is_ortho (U V : submodule 𝕜 E) : Prop :=
+U ≤ Vᗮ
+
+infix ` ⟂ `:50 := submodule.is_ortho
+
+lemma is_ortho.le {U V : submodule 𝕜 E} (h : U ⟂ V) : U ≤ Vᗮ := h
+
+lemma is_ortho_iff_le {U V : submodule 𝕜 E} : U ⟂ V ↔ U ≤ Vᗮ := iff.rfl
+
+@[symm]
+lemma is_ortho.symm {U V : submodule 𝕜 E} (h : U ⟂ V) : V ⟂ U :=
+(le_orthogonal_orthogonal _).trans (orthogonal_le h)
+lemma is_ortho_comm {U V : submodule 𝕜 E} : U ⟂ V ↔ V ⟂ U := ⟨is_ortho.symm, is_ortho.symm⟩
+lemma symmetric_is_ortho : symmetric (is_ortho : submodule 𝕜 E → submodule 𝕜 E → Prop) :=
+λ _ _, is_ortho.symm
+
+lemma is_ortho.inner_eq {U V : submodule 𝕜 E} (h : U ⟂ V) {u v : E} (hu : u ∈ U) (hv : v ∈ V) :
+  ⟪u, v⟫ = 0 :=
+h.symm hv _ hu
+
+lemma is_ortho_iff_inner_eq {U V : submodule 𝕜 E} : U ⟂ V ↔ ∀ (u ∈ U) (v ∈ V), ⟪u, v⟫ = 0 :=
+forall₄_congr $ λ u hu v hv, inner_eq_zero_symm
+
+/- TODO: generalize `submodule.map₂` to semilinear maps, so that we can state
+`U ⟂ V ↔ submodule.map₂ (innerₛₗ 𝕜) U V ≤ ⊥`. -/
+
+@[simp] lemma is_ortho_bot_left {V : submodule 𝕜 E} : ⊥ ⟂ V := bot_le
+@[simp] lemma is_ortho_bot_right {U : submodule 𝕜 E} : U ⟂ ⊥ := is_ortho_bot_left.symm
+
+lemma is_ortho.mono_left {U₁ U₂ V : submodule 𝕜 E} (hU : U₂ ≤ U₁) (h : U₁ ⟂ V) : U₂ ⟂ V :=
+hU.trans h
+
+lemma is_ortho.mono_right {U V₁ V₂ : submodule 𝕜 E} (hV : V₂ ≤ V₁) (h : U ⟂ V₁) : U ⟂ V₂ :=
+(h.symm.mono_left hV).symm
+
+lemma is_ortho.mono {U₁ V₁ U₂ V₂ : submodule 𝕜 E} (hU : U₂ ≤ U₁) (hV : V₂ ≤ V₁) (h : U₁ ⟂ V₁) :
+  U₂ ⟂ V₂ :=
+(h.mono_right hV).mono_left hU
+
+@[simp]
+lemma is_ortho_self {U : submodule 𝕜 E} : U ⟂ U ↔ U = ⊥ :=
+⟨λ h, eq_bot_iff.mpr $ λ x hx, inner_self_eq_zero.mp (h hx x hx), λ h, h.symm ▸ is_ortho_bot_left⟩
+
+@[simp] lemma self_is_ortho_orthogonal (U : submodule 𝕜 E) : U ⟂ Uᗮ :=
+le_orthogonal_orthogonal _
+
+@[simp] lemma orthogonal_is_ortho_self (U : submodule 𝕜 E) : Uᗮ ⟂ U :=
+(self_is_ortho_orthogonal U).symm
+
+@[simp]
+lemma is_ortho_top_right {U : submodule 𝕜 E} : U ⟂ ⊤ ↔ U = ⊥ :=
```

Comment:
Okay, this reads funny!

Reply:
I think I derived this name from `disjoint_top_right`, though I might be imagining things

Reply:
Yes, we more often care about its order properties than its algebraic ones (which only come in handy when doing pointwise operations).

Reply:
If these are really objectionable I could either write `is_ortho ⊥ V` or `(⊥ : submodule 𝕜 E) ⟂ V` here

Reply:
I think it's alright. My remark was inconsequential.